sdk/python/kfp/v2/dsl/type_utils.py

```diff
@@ -120,6 +125,7 @@ def get_input_artifact_type_schema(
     The input type schema if found in inputs, or None if not found.
   """
   for component_input in inputs:
-    if component_input.name == input_name:
+    if component_input.name == input_name and not is_parameter_type(
```

Comment:
Do you mind helping me understand why we need this condition (`is_parameter_type`) here?

I thought if we did the right thing in component bridge when adding to `task.inputs.artifacts`, we won't call this function with a parameter input, right?

Reply:
That's right. This is more of a safeguard for defensive programming. So this utility function itself can behave correctly even with possible misuse.

Reply:
Then can we actually `assert not is_parameter_type()` instead of swallowing the possible misuse?

Reply:
That works too. Is assert more suitable than raise in this caes?
Also, for consistency, I think we should do the same for not found case, as normally this shouldn't happen.

Reply:
`assert` usually catches the status which should not occur by design, while `raise` can be triggered by user input, or side effect.

So I guess `assert` works better here.